### PR TITLE
feature: Target transforms like Box-Cox (as in HEBO)

### PR DESCRIPTION
**Pull request by mseeger**
Next step towards HEBO. Allows for invertible target transform which can depend on a parameter. This parameter is learned together with all the others.

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: benchmarking/commons/benchmark_definitions/fcnet.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 def fcnet_benchmark(dataset_name):
     return SurrogateBenchmarkDefinition(
-        max_wallclock_time=1200,
+        max_wallclock_time=3600,
```
Comment (mseeger) [inline benchmarking/commons/benchmark_definitions/fcnet.py:20]: Also part of earlier PR, hope to get merged before. Increases time for fcnet, which is a bit short.

File: benchmarking/commons/hpo_main_simulator.py [hunk 2]
```diff
@@ -45,13 +45,13 @@
 from syne_tune.tuner import Tuner
```
Comment (mseeger) [inline benchmarking/commons/hpo_main_simulator.py:45]: Changes in these files are part of earlier PR, to get merged soon (hopefully)

File: syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/gluon_blocks_helpers.py [hunk 3]
```diff
@@ -103,7 +103,7 @@ class ScalarEncodingBase:
     ``dec`` is used in :meth:`set`. Use class:'IdentityScalarEncoding' for no
     encoding (identity). Note: ``enc`` (and ``dec``) must be strictly increasing.
 
-    Box constraints are given by ``constr_lower_int < constr_upper_int``.
+    Box constraints are given by ``constr_lower_int <= constr_upper_int``.
```
Comment (mseeger) [inline syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/gluon_blocks_helpers.py:106]: Should have been fixed earlier. Here, I need it to fix lambda at the beginning and only optimize it when there is enough data

File: syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/gp_regression.py [hunk 4]
```diff
@@ -27,9 +27,11 @@
     GaussianProcessMarginalLikelihood,
 )
 from syne_tune.optimizer.schedulers.searchers.bayesopt.gpautograd.mean import (
-    ScalarMeanFunction,
     MeanFunction,
 )
+from syne_tune.optimizer.schedulers.searchers.bayesopt.gpautograd.target_transform import (
+    ScalarTargetTransform,
```
Comment (wistuba) [inline syne_tune/optimizer/schedulers/searchers/bayesopt/gpautograd/gp_regression.py:33]: nit: remove comma and black will beautify correctly (unless the import is indeed that long)

**Review by wistuba — APPROVED** [review]

**Pull request merged**